Spatial.gd

```diff
@@ -816,77 +826,58 @@ func _on_NewPathPoint_pressed():
 		var z_accurate_player_position = player_position
 		z_accurate_player_position.z = -z_accurate_player_position.z
 		self.currently_active_path.add_point(z_accurate_player_position)
-		self.currently_active_path_2d.add_point(Vector2(self.player_position.x, -self.player_position.z))
+		self.currently_active_path2d.add_point(z_accurate_player_position)
 
 func _on_NodeEditorDialog_hide():
-	self.currently_selected_node = null
+	self.currently_selected_gizmo = null
 	clear_adjustment_nodes()
 	_on_Dialog_hide()
 
 
 func _on_DeleteNode_pressed():
-	if self.currently_selected_node.point_type == "icon":
-		self.currently_selected_node.object_link.get_parent().remove_child(self.currently_selected_node.object_link)
-		self.currently_selected_node.object_link.queue_free()
-	elif self.currently_selected_node.point_type == "path":
-		var path =   self.currently_selected_node.object_link
-		var path2d = self.currently_selected_node.object_2d_link
-		var index =  self.currently_selected_node.object_index
-
-		path.remove_point(index)
-		path2d.remove_point(index)
+	self.currently_selected_node3d.remove(self.currently_selected_index)
+	if self.currently_selected_node2d != null:
+		self.currently_selected_node2d.remove(self.currently_selected_index)
+	on_gizmo_deselected(self.currently_selected_gizmo)
 	clear_adjustment_nodes()
 	gen_adjustment_nodes()
-	on_gizmo_deselected(self.currently_selected_node)
 
 
 func _on_NewNodeAfter_pressed():
-	if self.currently_selected_node.point_type == "icon":
-		print("Warning: Cannot add node to icon")
-	elif self.currently_selected_node.point_type == "path":
-		print("insert path node")
-		var path = self.currently_selected_node.object_link
-		var path2d = self.currently_selected_node.object_2d_link
-		var index = self.currently_selected_node.object_index
-
-		var start = path.get_point_position(index)
-		var midpoint = self.player_position
-		midpoint.z = -midpoint.z
-		if path.get_point_count() > index+1:
-			var end = path.get_point_position(index+1)
-			midpoint = ((start-end)/2) + end
-
-		path.add_point(midpoint, index+1)
-		path2d.add_point(Vector2(midpoint.x, midpoint.z), index+1)
-
-		clear_adjustment_nodes()
-		gen_adjustment_nodes()
-		on_gizmo_deselected(self.currently_selected_node)
+	print("insert path node")
+	var midpoint = self.player_position
+	midpoint.z = -midpoint.z
+	self.currently_selected_node3d.new_point_after(midpoint, self.currently_selected_index)
```

Comment:
Ditto about currently_selected_node3d and currently_selected_gizmo being allowed to be null.

Reply:
I don't think currently_selected_gizmo can ever be null in this context. This button is only active if there is a selected gizmo and that gizmo is for a path. (Aside, I am going to change the name of the function to make this clearer) Should there still be null checks in that case?

Reply:
ditto the "should" vs "can" comment above.